docs/develop-component-checklist.md

```diff
@@ -0,0 +1,23 @@
+# Developing a component - things to remember
+
+##Don't forget..
```

Comment:
I feel like this is already covered in the [component conventions](https://github.com/alphagov/govuk_publishing_components/blob/master/docs/component_conventions.md#structure)?

Maybe we just need to expand on things there.